packages/artifact/src/internal/upload-http-client.ts

```diff
@@ -417,13 +423,13 @@ export class UploadHttpClient {
         core.info(
           `Backoff due to too many requests, retry #${retryCount}. Waiting for ${retryAfterValue} milliseconds before continuing the upload`
         )
-        await new Promise(resolve => setTimeout(resolve, retryAfterValue))
+        await sleep(retryAfterValue)
```

Comment:
If we implement exponential backoff inside `retry`, can we use `retryHttpClientResponse` here as well?

Reply:
See my earlier comment, right above in this code there already is exponential back-off, just a load of refactoring that I would like to hold-off for a future PR: https://github.com/actions/toolkit/pull/675#discussion_r545308637